homeassistant/components/sensor/zeversolar.py

```diff
@@ -0,0 +1,139 @@
+    url = (_RESOURCE + api_key)
+    var_conf = config.get(CONF_RESOURCES)
+
+    rest = ZeverzolarData(url)
+    rest.update()
+
+    sensors = []
+    for resource in var_conf:
+        sensors.append(ZeverzolarSensor(rest, name, resource))
+
+    add_devices(sensors, True)
+
+
+class ZeverzolarSensor(Entity):
+    """Implementation of a Zeversolar sensor."""
+
+    def __init__(self, rest, name, sensor_type):
+        """Initialize the sensor."""
+        self.rest = rest
+        self._name = name
+        self.type = sensor_type
+        self._state = STATE_UNKNOWN
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self._name is None:
+            return SENSOR_TYPES[self.type][0]
+        return '{} {}'.format(self._name, SENSOR_TYPES[self.type][0])
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self.rest.data is not None
+
+    @property
+    def state(self):
+        """Return the state of the resources."""
+        return self._state
+
+    def update(self):
+        """Get the latest data from REST API."""
+        self.rest.update()
+        value = self.rest.data
+
+        if value is not None:
+            if self.type == 'sid':
+                self._state = value['sid']
+            elif self.type == 'E-Today':
+                self._state = value['E-Today']['value']
+                self._unit_of_measurement = value['E-Today']['unit']
+            elif self.type == 'E-Month':
+                self._state = value['E-Month']['value']
+                self._unit_of_measurement = value['E-Month']['unit']
+            elif self.type == 'E-Total':
+                self._state = value['E-Total']['value']
+                self._unit_of_measurement = value['E-Total']['unit']
+            elif self.type == 'TotalYield':
+                self._state = value['TotalYield']['value']
+                self._unit_of_measurement = value['TotalYield']['unit']
+            elif self.type == 'CO2Avoided':
+                self._state = value['CO2Avoided']['value']
+                self._unit_of_measurement = value['CO2Avoided']['unit']
+            elif self.type == 'Power':
+                self._state = value['Power']['value']
+                self._unit_of_measurement = value['Power']['unit']
+
+
+class ZeverzolarData:
+    """Get data from ZeverSolar."""
+
+    def __init__(self, resource):
+        """Initialize the data object."""
+        self._resource = resource
+        self.data = dict()
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Get the latest data from the Zeversolar REST API."""
+        try:
+            response = requests.get(self._resource, timeout=10)
```

Comment:
All of the rest logic and unpacking the api response json should occur in a separate pypi package that's imported by this component. See: https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devices-services for more details

Reply:
@rohankapoorcom I did not knew that, I will close this PR.
Since I have to rewrite a lot of code according to the new style.
Thank you for your awareness.